src/TestFramework/TestFramework/Assertions/Assert.AreSame.cs

```diff
@@ -10,6 +12,120 @@ namespace Microsoft.VisualStudio.TestTools.UnitTesting;
 /// </summary>
 public sealed partial class Assert
 {
+    [InterpolatedStringHandler]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public readonly struct AssertAreSameInterpolatedStringHandler<TArgument>
+    {
+        private readonly StringBuilder? _builder;
+        private readonly TArgument? _expected;
+        private readonly TArgument? _actual;
+
+        public AssertAreSameInterpolatedStringHandler(int literalLength, int formattedCount, TArgument? expected, TArgument? actual, out bool shouldAppend)
+        {
+            _expected = expected;
+            _actual = actual;
+            shouldAppend = IsAreSameFailing(expected, actual);
+            if (shouldAppend)
+            {
+                _builder = new StringBuilder(literalLength + formattedCount);
+            }
+        }
+
+        internal void ComputeAssertion()
+        {
+            if (_builder is not null)
+            {
+                ThrowAssertAreSameFailed(_expected, _actual, _builder.ToString());
+            }
+        }
+
+        public void AppendLiteral(string value) => _builder!.Append(value);
+
+        public void AppendFormatted<T>(T value) => AppendFormatted(value, format: null);
```

Comment:
I should be adding `AppendFormatted(string?)` overload. This should be resolved before merging.